test/controllers/api/v2/host_tracer_controller_test.rb

```diff
@@ -28,53 +28,25 @@ def test_index
       assert_includes results['results'].collect { |item| item['id'] }, @host1.host_traces.first.id
     end
 
-    def test_resolve_group_by_helper
+    def test_resolve
       trace_one = Katello::HostTracer.create(host_id: @host1.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
-      trace_two = Katello::HostTracer.create(host_id: @host2.id, application: 'rsyslog', app_type: 'daemon', helper: 'systemctl restart rsyslog')
-      helper = {:helper => trace_two.helper}
+      task_output = {"id" => nil, "label" => nil, "started_at" => nil, "ended_at" => nil, "state" => nil, "result" => nil, "external_id" => nil,
+                     "parent_task_id" => nil, "start_at" => nil, "start_before" => nil, "action" => "N/A N/A", "user_id" => nil, "state_updated_at" => nil}
 
-      job_invocation = {"description" => "Restart Services", "id" => 1, "job_category" => "Katello"}
-      JobInvocationComposer.expects(:for_feature).with(:katello_service_restart, [@host1.id, @host2.id], helper).returns(mock(trigger: true, job_invocation: job_invocation))
+      job_invocation = mock(task_id: '12345')
+      composer = mock(job_invocation: job_invocation)
 
-      put :resolve, params: { use_route: 'traces/resolve', :trace_ids => [trace_one.id, trace_two.id]}
+      ForemanTasks::Task.stubs(:find).returns(ForemanTasks::Task.new)
 
-      assert_response :success
-
-      response_body = JSON.parse(response.body)
-      assert_equal response_body, [job_invocation]
-    end
+      Katello::HostTraceManager.expects(:resolve_traces).with([trace_one]).returns([composer])
 
-    def test_resolve_reboot_service
```

Comment:
This test and the others which test the grouping are now behaviors of the HostTracesManager, so these tests should be moved to a file which tests that class. Good new is since we're not dealing with any controller stuff, they become even simpler

Reply:
Yep, moved the 3 main tests to the service class tests, and kept 1 for the main resolve and 1 for the bulk resolve test.